src/tests/test_contracts/test_yas_pool.cairo

```diff
@@ -29,22 +31,178 @@ mod YASPoolTests {
-            let token_0 = contract_address_const::<'TOKEN0'>();
-            let token_1 = contract_address_const::<'TOKEN1'>();
             let fee = 5;
             let tick_spacing = IntegerTrait::<i32>::new(1, false);
-            let yas_pool = deploy(factory, token_0, token_1, fee, tick_spacing);
+            let yas_pool = deploy(FACTORY_ADDRESS(), TOKEN_A(), TOKEN_B(), fee, tick_spacing);
+        }
+    }
+
+    mod Initialize {
+        use super::deploy;
+
+        use starknet::testing::pop_log;
+
+        use yas::contracts::yas_pool::{
+            IYASPoolDispatcherTrait, YASPool::{YASPoolImpl, InternalImpl, Initialize, Slot0}
+        };
+        use yas::numbers::fixed_point::implementations::impl_64x96::{
+            FP64x96Impl, FP64x96Sub, FP64x96PartialEq, FixedType, FixedTrait
+        };
+        use yas::numbers::signed_integer::{i32::i32, integer_trait::IntegerTrait};
+        use yas::libraries::tick_math::TickMath::{MAX_SQRT_RATIO, MIN_SQRT_RATIO};
+        use yas::utils::{math_utils::pow, utils::Slot0PartialEq};
+        use yas::tests::utils::constants::PoolConstants::{
+            FACTORY_ADDRESS, TOKEN_A, TOKEN_B, STATE, min_tick, max_tick
+        };
+
+        #[test]
+        #[available_gas(200000000)]
+        #[should_panic(expected: ('AI', 'ENTRYPOINT_FAILED'))]
+        fn test_fails_if_already_initialized() {
+            let yas_pool = deploy(
+                FACTORY_ADDRESS(), TOKEN_A(), TOKEN_B(), 5, IntegerTrait::<i32>::new(1, false)
+            );
+
+            // sqrt_price_X96 is the result of encode_price_sqrt(1, 1) on v3-core typescript impl. 
+            let sqrt_price_X96 = FixedTrait::new(79228162514264337593543950336, false);
+            yas_pool.initialize(sqrt_price_X96);
+
+            // initialize again with same values
+            yas_pool.initialize(sqrt_price_X96);
+        }
+
+        #[test]
+        #[available_gas(200000000)]
+        #[should_panic(expected: ('R', 'ENTRYPOINT_FAILED'))]
+        fn test_fails_if_price_is_too_low() {
+            let yas_pool = deploy(
+                FACTORY_ADDRESS(), TOKEN_A(), TOKEN_B(), 5, IntegerTrait::<i32>::new(1, false)
+            );
+
+            let sqrt_price_X96 = FixedTrait::new(1, false);
+            yas_pool.initialize(sqrt_price_X96);
+        }
+
+        #[test]
+        #[available_gas(200000000)]
+        #[should_panic(expected: ('R', 'ENTRYPOINT_FAILED'))]
+        fn test_fails_if_price_is_min_sqrt_ratio_minus_1() {
+            let yas_pool = deploy(
+                FACTORY_ADDRESS(), TOKEN_A(), TOKEN_B(), 5, IntegerTrait::<i32>::new(1, false)
+            );
+
+            let sqrt_price_X96 = FixedTrait::new(MIN_SQRT_RATIO - 1, false);
+            yas_pool.initialize(sqrt_price_X96);
+        }
+
+        #[test]
+        #[available_gas(200000000)]
+        #[should_panic(expected: ('R', 'ENTRYPOINT_FAILED'))]
+        fn test_fails_if_price_is_too_high() {
+            let yas_pool = deploy(
```

Comment:
What's the reasoning behind testing with deploy here and not just using the State?